sympy/core/tests/test_sorting.py

```diff
@@ -26,3 +27,9 @@ def test_ordered():
     assert list(ordered(l, warn=True)) == [[1], [1], [2]]
     raises(ValueError, lambda: list(ordered(['a', 'ab'], keys=[lambda x: x[0]],
         default=False, warn=True)))
+
+
+def test_tie_breaker_in_case_of_equality():
+    # Issue #24045
+    a, b, c = symbols("a b c")
+    powsimp(exp(a)/((c*a - c*b)*(Float(1.0)*c*a - Float(1.0)*c*b)))
```

Comment:
A better (more direct) test for the actual problem is this:
```python
In [7]: a, b, c = symbols("a b c")

In [8]: list(ordered([a*c - b*c, 1.0*a*c - 1.0*b*c]))
---------------------------------------------------------------------------
TypeError 
```
That should be tested directly rather than indirectly via `powsimp`.